testing/test_pluginmanager.py

```diff
@@ -447,64 +447,40 @@ def example_hook():
 
 
 def test_load_setuptools_instantiation(monkeypatch, pm):
-    pkg_resources = pytest.importorskip("pkg_resources")
+    class EntryPoint(object):
+        name = "myname"
+        group = "hello"
+        value = "myname:foo"
 
-    def my_iter(group, name=None):
-        assert group == "hello"
+        def load(self):
+            class PseudoPlugin(object):
+                x = 42
 
-        class EntryPoint(object):
-            name = "myname"
-            dist = None
+            return PseudoPlugin()
 
-            def load(self):
-                class PseudoPlugin(object):
-                    x = 42
+    class Distribution(object):
+        entry_points = (EntryPoint(),)
 
-                return PseudoPlugin()
+    dist = Distribution()
 
-        return iter([EntryPoint()])
+    def my_distributions():
+        return (dist,)
 
-    monkeypatch.setattr(pkg_resources, "iter_entry_points", my_iter)
+    monkeypatch.setattr(importlib_metadata, "distributions", my_distributions)
     num = pm.load_setuptools_entrypoints("hello")
     assert num == 1
     plugin = pm.get_plugin("myname")
     assert plugin.x == 42
-    assert pm.list_plugin_distinfo() == [(plugin, None)]
+    ret = pm.list_plugin_distinfo()
+    # poor man's `assert ret == [(plugin, mock.ANY)]`
+    assert len(ret) == 1
+    assert len(ret[0]) == 2
+    assert ret[0][0] == plugin
+    assert ret[0][1]._dist == dist
     num = pm.load_setuptools_entrypoints("hello")
     assert num == 0  # no plugin loaded by this call
 
 
-def test_load_setuptools_version_conflict(monkeypatch, pm):
```

Comment:
runtime checking of dependency versions is not possible without the old and slow pkg_resources? ;-(

Reply:
hmmm I guess that's not the way I saw that.  any time I've dealt with `pkg_resources` the side-effect of looking up package metadata triggering `VersionConflict` I always saw as a nuisance.  It's not really protecting you against anything, you're already in the broken state.

It would be possible to do with importlib-metadata, but traversing the dependency graph is wasted cpu cycles :)